src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Emitter/CoreBindingHelpers.cs

```diff
@@ -901,6 +901,15 @@ private bool EmitBindImplForMember(
                         }
                     case ComplexTypeSpec complexType:
                         {
+                            // Early detection of types we cannot bind to and skip it.
+                            if (!_typeIndex.HasBindableMembers(complexType) &&
+                                !_typeIndex.GetEffectiveTypeSpec(complexType).IsValueType &&
+                                complexType is not CollectionSpec &&
+                                (complexType as ObjectSpec).InstantiationStrategy == ObjectInstantiationStrategy.ParameterizedConstructor)
```

Comment:
```suggestion
                                ((ObjectSpec)complexType).InstantiationStrategy == ObjectInstantiationStrategy.ParameterizedConstructor)
```

This way we get a cast exception (vs. a nullref) if this ever isn't the right type.